proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/EidasAuthnRequestResource.java

```diff
@@ -78,11 +75,14 @@ private View handleAuthnRequest(String encodedEidasAuthnRequest, String eidasRel
     }
 
     private void setResponseDataInSession(HttpSession session, EidasSamlParserResponse eidasSamlParserResponse, AuthnRequestResponse vspResponse, String eidasRelayState) {
-        session.setAttribute(SESSION_KEY_EIDAS_REQUEST_ID, eidasSamlParserResponse.getRequestId());
-        session.setAttribute(SESSION_KEY_EIDAS_CONNECTOR_PUBLIC_CERT, eidasSamlParserResponse.getConnectorEncryptionPublicCertificate());
-        session.setAttribute(SESSION_KEY_EIDAS_DESTINATION, eidasSamlParserResponse.getDestination());
-        session.setAttribute(SESSION_KEY_EIDAS_RELAY_STATE, eidasRelayState);
-        session.setAttribute(SESSION_KEY_HUB_REQUEST_ID, vspResponse.getRequestId());
+        GatewaySessionData gatewaySessionData = new GatewaySessionData(
```

Comment:
I think it would be better to pass the `vspResponse`, `eidasSamlParserResponse` and `eidasRelayState` as objects and extract the values in `GatewaySessionData` rather than extracting the strings here.

Reply:
Yeah, that'd be nicer, I'll fix it up.